### DISCUSSION TITLE
Granular Data Collection

### DISCUSSION THREAD

**Discussion by paulmccrady** [cloudyspells/PSRule.Rules.AzureDevOps#46] [Ideas]
At present the data collection part of the 'Export' functions call a number of APIs that the user has no control over. Some of these APIs require Full Access PAT permissions. Can we include a feature that allows the user/caller to choose whether to include data collection that requires tokens with such large/sensitive/destructive permissions sets ?

**Comment by webtonize** [comment]
Thanks for your idea. As we discussed in the _Q&A_ section I really like the idea of using decent controlled and documented permission sets. I have opened issue #47 and plan to start work on it soon. Related to this I have also opened #48 to improve export coverage without excessive permissions.

**Comment by webtonize** [comment]
[v0.2.0](https://github.com/cloudyspells/PSRule.Rules.AzureDevOps/releases/tag/v0.2.0) has just been released. In this release is a new feature to specify the level of access available for the PAT through a `TokenType` parameter. This supports 3 levels of access ranging from `ReadOnly` to `FullAccess`.
